diff --git a/node_modules/gatsby/cache-dir/find-path.js b/node_modules/gatsby/cache-dir/find-path.js
index a231735..5dd8f90 100644
--- a/node_modules/gatsby/cache-dir/find-path.js
+++ b/node_modules/gatsby/cache-dir/find-path.js
@@ -109,6 +109,35 @@ export const grabMatchParams = rawPathname => {
   return {}
 }
 
+function getCookies() {
+  var c = document.cookie, v = 0, cookies = {};
+  if (document.cookie.match(/^\s*\$Version=(?:"1"|1);\s*(.*)/)) {
+      c = RegExp.$1;
+      v = 1;
+  }
+  if (v === 0) {
+      c.split(/[,;]/).map(function(cookie) {
+          var parts = cookie.split(/=/, 2),
+              name = decodeURIComponent(parts[0].trimLeft()),
+              value = parts.length > 1 ? decodeURIComponent(parts[1].trimRight()) : null;
+          cookies[name] = value;
+      });
+  } else {
+      c.match(/(?:^|\s+)([!#$%&'*+\-.0-9A-Z^`a-z|~]+)=([!#$%&'*+\-.0-9A-Z^`a-z|~]*|"(?:[\x20-\x7E\x80\xFF]|\\[\x00-\x7F])*")(?=\s*[,;]|$)/g).map(function($0, $1) {
+          var name = $0,
+              value = $1.charAt(0) === '"'
+                        ? $1.substr(1, -1).replace(/\\(.)/g, "$1")
+                        : $1;
+          cookies[name] = value;
+      });
+  }
+  return cookies;
+}
+
+function getCookie(name) {
+  return getCookies()[name];
+}
+
 // Given a raw URL path, returns the cleaned version of it (trim off
 // `#` and query params), or if it matches an entry in
 // `match-paths.json`, its matched path is returned
@@ -134,6 +163,31 @@ export const findPath = rawPathname => {
     foundPath = cleanPath(rawPathname)
   }
 
+  // This matches the initial call to `findPath` where
+  // `rawPathname === /` and we need to determine the
+  // rewrite based on the user's cookie value.
+  if (rawPathname === '/') {
+    const cv = getCookie('test_bucket')
+    if (cv === 'a') {
+      foundPath = '/home-a'
+    }
+    if (cv === 'b') {
+      foundPath = '/home-b'
+    }
+  }
+
+  // Once we match interally we get another call to `findPath`
+  // where `rawPathname === /home-a` or `rawPathname === /home-b`
+  // depending on the cookie value. This makes sure to catch that
+  // case as well.
+  if (rawPathname === '/home-a') {
+    foundPath = '/home-a'
+  }
+
+  if (rawPathname === '/home-b') {
+    foundPath = '/home-b'
+  }
+
   pathCache.set(trimmedPathname, foundPath)
 
   return foundPath
